src/test/kotlin/com/terraformation/backend/accelerator/ModuleEventSearchTest.kt

```diff
@@ -0,0 +1,286 @@
+
+    val expected =
+        SearchResults(
+            listOf(
+                mapOf("id" to "$event3"),
+                mapOf("id" to "$event1"),
+                mapOf("id" to "$event2"),
+            ),
+            null)
+
+    val actual = searchService.search(prefix, fields, NoConditionNode())
+
+    assertJsonEquals(expected, actual)
+  }
+
+  @Test
+  fun `can search for parent module`() {
+    val module = inserted.moduleId
+    val event = insertEvent(moduleId = module)
+
+    val otherModule = insertModule()
+    val otherEvent = insertEvent(moduleId = otherModule)
+
+    val prefix = SearchFieldPrefix(searchTables.events)
+    val fields = listOf("id", "module.id").map { prefix.resolve(it) }
+    val expected =
+        SearchResults(
+            listOf(
+                mapOf("id" to "$event", "module" to mapOf("id" to "$module")),
+                mapOf("id" to "$otherEvent", "module" to mapOf("id" to "$otherModule"))),
+            null)
+
+    val actual = searchService.search(prefix, fields, NoConditionNode())
+
+    assertJsonEquals(expected, actual)
+  }
+
+  @Test
+  fun `can search for associated projects`() {
+    val project1 = insertProject(participantId = inserted.participantId)
+    val project2 = insertProject(participantId = inserted.participantId)
+    val project3 = insertProject(participantId = inserted.participantId)
+    val event1 = insertEvent()
+    insertEventProject(event1, project1)
+    insertEventProject(event1, project2)
+    insertEventProject(event1, project3)
+
+    val project4 = insertProject(participantId = inserted.participantId)
+    val project5 = insertProject(participantId = inserted.participantId)
+    val project6 = insertProject(participantId = inserted.participantId)
+    val event2 = insertEvent()
+    insertEventProject(event2, project4)
+    insertEventProject(event2, project5)
+    insertEventProject(event2, project6)
+
+    val prefix = SearchFieldPrefix(searchTables.events)
+    val fields = listOf("id", "projects.id").map { prefix.resolve(it) }
+
+    val expected =
+        SearchResults(
+            listOf(
+                mapOf(
+                    "id" to "$event1",
+                    "projects" to
+                        listOf(
+                            mapOf("id" to "$project1"),
+                            mapOf("id" to "$project2"),
+                            mapOf("id" to "$project3")),
+                ),
+                mapOf(
+                    "id" to "$event2",
+                    "projects" to
+                        listOf(
+                            mapOf("id" to "$project4"),
+                            mapOf("id" to "$project5"),
+                            mapOf("id" to "$project6")),
+                ),
+            ),
+            null)
+
+    val actual = searchService.search(prefix, fields, NoConditionNode())
+
+    assertJsonEquals(expected, actual)
+  }
+
+  @Test
+  fun `can search for events by projects`() {
```

Comment:
What this test is checking for is good. When I read this test name, I though the test was going to check whether you could have a _filter_ with a project ID, that is, flipping the structure of this and searching on the events table with a `FieldNode` that searches `projects.id`.

I think it'd be good to include that as well, again because this is introducing a new pattern for the many-to-many relationships and we should verify that the logic for filtering across this kind of relationship works as expected, in addition to verifying the logic for fetching across the relationship (which this test and the others in the PR already cover nicely).